### DISCUSSION TITLE
How is the mysql adaptation? More data sources! The amazing malloy !!

### DISCUSSION THREAD

**Discussion by PeatBoy** [malloydata/malloy#2049] [General]
Hi, the developer of malloy, it's a great honor to explore your Malloy project, which is a very bright DSL -> SQL implementation. Although it is in the experimental device at present, I am very optimistic about its potential!

I want to ask:
1. How long will MySQL's adaptation progress stabilize?
2. Can I provide duckDB or presto to make malloy indirectly support mysql queries?
3. Are there any adaptation plans for more data sources? Personally, I need doris, Clickhouse, etc.
4. Does the official provide the method implementation of SQL -> malloy?

I look forward to your reply and join the adaptation work.

**Comment by mtoy-googly-moogly** [comment]
MySQL is a supported dialect now. We have a few users and they seem content. The MySQL dialect of SQL does have some limitations, but it is mostly Malloy and seems to work well.

We are not currently working on any other new dialects. We know of a few projects in existence where people are working on pull requests to expand the supported dialect list. Maybe if you asked in slack you could find someone working on those additional dialects.

**Reply by aszenz** [reply]
Thats good to know, I have been using MySQL dialect and it mostly works yes but I have run into some issues: 
- In some scenarios nests throw syntax errors likely when the source includes joins
- In another scenario malloy seemingly forgot a join relation and threw a syntax error. 

I'll try to write a reproducer for these if I know it would get attention
